From 159dff274a7f32f9b53711df6a5b6fbae8c64b4a Mon Sep 17 00:00:00 2001
From: Martin Negyokru <negyokru@inf.u-szeged.hu>
Date: Thu, 7 Dec 2023 15:37:53 +0100
Subject: [PATCH] Fix building with system libxml2

Libxml2 2.12.0 has changed the const-ness of the
xmlError pointers, which results in a build error.

Pick-to: 112-based 108-based 87-based
Change-Id: Id86fcfbfe004f89109808b9bdb11c01133553884
Reviewed-on: https://codereview.qt-project.org/c/qt/qtwebengine-chromium/+/523633
Reviewed-by: Allan Sandfeld Jensen <allan.jensen@qt.io>
---
 .../third_party/blink/renderer/core/xml/xslt_processor.h     | 5 +++++
 .../blink/renderer/core/xml/xslt_processor_libxslt.cc        | 4 ++++
 2 files changed, 9 insertions(+)

diff --git a/src/3rdparty/chromium/third_party/blink/renderer/core/xml/xslt_processor.h b/src/3rdparty/chromium/third_party/blink/renderer/core/xml/xslt_processor.h
index d53835e9675..72536e4fd7d 100644
--- a/src/3rdparty/chromium/third_party/blink/renderer/core/xml/xslt_processor.h
+++ b/src/3rdparty/chromium/third_party/blink/renderer/core/xml/xslt_processor.h
@@ -79,7 +79,12 @@ class XSLTProcessor final : public ScriptWrappable {

   void reset();

+#if LIBXML_VERSION >= 21200
+  static void ParseErrorFunc(void* user_data, const xmlError*);
+#else
   static void ParseErrorFunc(void* user_data, xmlError*);
+#endif
+
   static void GenericErrorFunc(void* user_data, const char* msg, ...);

   // Only for libXSLT callbacks
diff --git a/src/3rdparty/chromium/third_party/blink/renderer/core/xml/xslt_processor_libxslt.cc b/src/3rdparty/chromium/third_party/blink/renderer/core/xml/xslt_processor_libxslt.cc
index 133e0b3355d..e8e6a09f485 100644
--- a/src/3rdparty/chromium/third_party/blink/renderer/core/xml/xslt_processor_libxslt.cc
+++ b/src/3rdparty/chromium/third_party/blink/renderer/core/xml/xslt_processor_libxslt.cc
@@ -60,7 +60,11 @@ void XSLTProcessor::GenericErrorFunc(void*, const char*, ...) {
   // It would be nice to do something with this error message.
 }

+#if LIBXML_VERSION >= 21200
+void XSLTProcessor::ParseErrorFunc(void* user_data, const xmlError* error) {
+#else
 void XSLTProcessor::ParseErrorFunc(void* user_data, xmlError* error) {
+#endif
   FrameConsole* console = static_cast<FrameConsole*>(user_data);
   if (!console)
     return;
-- 
2.44.0

